sky/execution.py

```diff
@@ -529,6 +529,13 @@ def launch(
             ]
             skip_unnecessary_provisioning = True
 
+    detach_setup = True
+    if controller_utils.Controllers.from_name(cluster_name) is not None:
```

Comment:
nit: can we just do `detach_setup = controller_utils.Controllers.from_name(cluster_name) is not None`?